src/test/java/seedu/planitarium/command/AddRecordTest.java

```diff
@@ -1,4 +1,129 @@
+            "Unknown error is detected from 'Invalid category index `-1`', please check again.";
+    protected static final String INVALID_GROUPINX_ERROR_MSG =
+            "Unknown error is detected from 'Invalid group index `5`', please check again.";
+
+    Family family;
+    AddRecordCommand addRec;
+    AddRecordCommand addIn;
+    AddRecordCommand addOut;
+
+    private void initialize() throws PlanITariumException {
+        family = new Family();
+        AddPersonCommand add = new AddPersonCommand(CommandsForTesting.ADDPERSON, family);
+        add.execute();
+    }
+
+    @Test
+    void addRec_invalidUid_fail() {
+        try {
+            initialize();
+            addRec = new AddRecordCommand(CommandsForTesting.ADDINCOME2, family);
+        } catch (PlanITariumException e) {
+            assertEquals(e.toString(), INVALID_UID_MSG);
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    void addIncome_invalidMoney_fail() {
+        try {
+            initialize();
+            addRec = new AddRecordCommand(CommandsForTesting.ADDINCOME3, family);
+        } catch (PlanITariumException e) {
+            assertEquals(e.toString(), INVALID_IMONEY_MSG2);
+        } catch (Exception e) {
+            fail();
+        }
+
+        try {
+            initialize();
+            addRec = new AddRecordCommand(CommandsForTesting.ADDINCOME4, family);
+        } catch (PlanITariumException e) {
+            assertEquals(e.toString(), INVALID_IMONEY_MSG);
+        } catch (Exception e) {
+            fail();
+        }
+
+        try {
+            initialize();
+            addRec = new AddRecordCommand(CommandsForTesting.ADDEXPEND2, family);
+        } catch (PlanITariumException e) {
+            assertEquals(e.toString(), INVALID_EMONEY_MSG);
+        } catch (Exception e) {
+            fail();
+        }
+
+    }
+
+    @Test
+    void addPerson_invalidGroupInx_fail() {
+        try {
+            addRec = new AddRecordCommand(CommandsForTesting.ADDINCOME5, family);
+        } catch (PlanITariumException e) {
+            assertEquals(e.toString(), INVALID_GROUPINX_ERROR_MSG);
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    void find_invalidDes_fail() {
+        try {
+            initialize();
+            addIn = new AddRecordCommand(CommandsForTesting.ADDINCOME6, family);
+        } catch (PlanITariumException e) {
+            assertEquals(e.toString(), INVALID_DES_MSG);
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    void find_invalidCatInx_fail() {
+        try {
+            initialize();
+            addOut = new AddRecordCommand(CommandsForTesting.ADDEXPEND3, family);
+        } catch (PlanITariumException e) {
+            assertEquals(e.toString(), INVALID_CAT_MSG);
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    void addRec_success() {
```

Comment:
Did you forget to specify the type of inputs that `AddRecordCommand` will be given?